support-workers/src/main/scala/com/gu/support/workers/lambdas/CreateZuoraSubscription.scala

```diff
@@ -34,8 +38,8 @@ class CreateZuoraSubscription(servicesProvider: ServiceProvider = ServiceProvide
     services: Services
   ): FutureHandlerResult = {
     val now = () => OffsetDateTime.now
-    val subscribeItem = buildSubscribeItem(state, services.promotionService)
     for {
+      subscribeItem <- Future.fromTry(buildSubscribeItem(state, services.promotionService).left.map(err => new BuildSubscribePromoError(err)).toTry)
```

Comment:
a bit of gymnasitcs needed to get the Either into a Future, not sure if a nicer way